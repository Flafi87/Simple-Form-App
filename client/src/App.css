.error {
  color: red;
}
.form-group {
  height: 70px;
}
