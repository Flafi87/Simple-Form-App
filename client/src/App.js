import React from 'react';
import './App.css';

import { connect } from 'react-redux';

import 'bootstrap/dist/css/bootstrap.min.css';
import { Container, Row } from 'react-bootstrap';
import PropTypes from 'prop-types';
import addEvent from './actions/eventAction';
import FormPage from './components/FormPage';

function App({ addEvent, error }) {
  const submit = (values) => {
    addEvent(values);
  };
  return (
    <Container>
      <div className="App">
        <Row className="justify-content-md-center">
          <header className="App-header">
            <h1>Simple Form App</h1>
          </header>
        </Row>
        <FormPage onSubmit={submit} />
        {error ? <div className="error">{error}</div> : null}
      </div>
    </Container>
  );
}

App.propTypes = {
  addEvent: PropTypes.func.isRequired,
  error: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  error: state.errors.postError,
});


export default connect(
  mapStateToProps,
  { addEvent },
)(App);
