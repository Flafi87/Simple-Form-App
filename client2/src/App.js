import React from 'react';
import './App.css';
import Form from './components/Form'
import { connect } from 'react-redux';
import { addEvent } from './actions/eventAction'

function App({ error, addEvent }) {
  const submit = (values) => {
    addEvent(values)
  }
  return (
    <div className="App">
      <header className="App-header">
        <h1>Simple Form App</h1>
      </header>
      <Form onSubmit={submit} />
    </div>
  );
}

const mapStateToProps = (state) => ({
  error: state.form,
});

export default connect(
  mapStateToProps,
  { addEvent },
)(App);
