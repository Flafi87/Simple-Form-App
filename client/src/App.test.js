import React from 'react'
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import App from './App';
const mockStore = configureStore([]);
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});