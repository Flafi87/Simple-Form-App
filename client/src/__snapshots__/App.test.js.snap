// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Simple Form App
    </h1>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <label>
          First Name
        </label>
        <div>
          <input
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div>
        <label>
          Last Name
        </label>
        <div>
          <input
            name="secondName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div>
        <label>
          Email
        </label>
        <div>
          <input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div>
        <label>
          Event Date
        </label>
        <div>
          <input
            name="eventDate"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Event Date"
            type="date"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
